lib/coinbase/staking_reward.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module Coinbase
+  # A representation of a staking reward earned on a network for a given asset.
+  class StakingReward
+    # Returns a list of StakingRewards for the provided network, asset, and addresses.
+    # @param network_id [Symbol] The network ID
+    # @param asset_id [Symbol] The asset ID
+    # @param address_ids [Array<String>] The address IDs
+    # @param start_time [Time] The start time
+    # @param end_time [Time] The end time
+    # @param format [Symbol] The format to return the rewards in. (:usd, :native) Defaults to :usd.
+    # @return [Enumerable<Coinbase::StakingReward>] The staking rewards
+    def self.list(network_id, asset_id, address_ids, start_time, end_time, format: :usd)
```

Comment:
This is a ton of arguments, I wonder if this would be more readable with all keyword arguments?

Is there a default start_time / end_time that we can use potentially?

Reply:
Not a blocker, but just want to make this as easy to reason about as possible!

I think that this could be something like the following when added to the `wallet` object:

```
wallet.staking_rewards(:eth)
wallet.staking_rewards(:eth, start_time: Time.now -24*60*6), end_time: Time.now)
```

But on this endpoint since we need to provide a bunch of arguments, could all be keyword?

Reply:
I was thinking of doing a start time, but it seems odd, I think end_time as today would be a great default though

Reply:
I added both! (1 month previous -> now)